File: src/Abstract/MainObject.java
package Abstract;

interface IStudentState {
    void working();
    void going();
    void relaxing();
}

public class MainObject implements IStudentState {
    @Override
    public void working() {
        System.out.println("Student is working too hard doing math!");;
    }

    @Override
    public void going() {
        System.out.println("Student is going to the gym!");
    }

    @Override
    public void relaxing() {
        System.out.println("Student is watching movie in the cinema!");
    }
}

File: src/Abstract/MainObject.cpp
#include <iostream>
using namespace std;
class StudentState {
public:
	virtual void working();
	virtual void going();
	virtual void relaxing();
};

class StudentStateRealization : public StudentState {
public:
	void working() {
		cout << "Student is working too hard doing math!";
	}

	void going() {
		cout << "Student is going to the gym!";
	}

	void relaxing() {
		cout << "Student is watching movie in the cinema!";
	}
};